Client/Frontend/Browser/HashchangeHelper.swift

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+
+protocol HashchangeHelperDelegate: class {
+    func hashchangeHelperDidHashchange(hashchangeHelper: HashchangeHelper)
+}
+
+class HashchangeHelper: NSObject, BrowserHelper, UIGestureRecognizerDelegate {
```

Comment:
Also, you can remove the `UIGestureRecognizerDelegate` since this helper doesn't do anything with gestures.